cmd/juju/model/configcommand.go

```diff
@@ -155,7 +221,10 @@ func (c *configCommand) handleZeroArgs() error {
 func (c *configCommand) handleOneArg(arg string) error {
 	// We may have a single config.yaml file
 	_, err := os.Stat(arg)
-	if err == nil || strings.Contains(arg, "=") {
+	if err == nil {
+		// Handle yaml files
+		return c.parseYAMLFile(arg)
+	} else if strings.Contains(arg, "=") {
```

Comment:
Do we need an `else if !os.IsNotExist(err) { return errors.Trace(err) }` in case that you pass a file path that you don't have permission to access?

Reply:
Fine.

Reply:
Can't do this, as it falls through. We would have to change a lot of logic to fix this and I'm not sure it's worth it.

Reply:
Why not? Wouldn't this do the trick?

```go
var yamlSrc io.Reader
if arg == "-" {
  yamlSrc = os.Stdin
} else if _, err := os.Stat(arg); err != nil {
   if os.IsNotExist(err) && strings.Contains(arg, "=") {
     return c.parseSetKeys([]string{arg})
  }

  // Operator passed a file that we are not allowed to read 
 return errors.Trace(err)
} else {
  f, err := os.Open(arg)
  if err != nil {...}
 defer func(){ _ = f.Close() }()
 yamlSrc = f
}

// Pass yamlSrc to a method that reads and processes yaml data.
```